elytron/src/main/java/org/wildfly/extension/elytron/SSLContextResource.java

```diff
@@ -129,6 +130,9 @@ public Resource clone() {
      */
     private boolean hasActiveSessions() {
         final SSLContext sslContext = getSSLContext(sslContextServiceController);
+        if (sslContext instanceof DynamicSSLContext) {
```

Comment:
What about the case where the module that provides this class doesn't exist?

Reply:
If so, the best approach is to actually include this dynamic-ssl module in product but document it as not supported module. And the resource itself won't be possible to register in DEFAULT stability. This sounds better than removing these calls in wildfly-to-product chain

Reply:
Or should I use classloading and ServiceLoader here and in other instances in wildfly-core?

Reply:
@pferraro Do you have any thoughts on this?

Reply:
Can we make DynamicSSLContext extend another interface in Elytron that will not be excluded, we can then do the instance check based on that.

Reply:
Those are only going to be used at Stage.RUNTIME once this feature is enabled AND a resource is configured.

I think we can move those calls into a class with a method that can return the instance and the SSLDefininitions class updated to call that directly.

Reply:
@darranl Okay I'll be working on that today and will post an update here

Reply:
@Skyllarr I think the part that needs changes in WildFly Elytron is the higher priority to be included in the release from @fjuma on the WildFly (and core) side if the rest of the PR is approved I think we can get away with a blocker Jira issue to follow up.  There are also changed coming to Galleon that we may be able to take advantage of after Beta to automate the provisiononing of this artifact based on stability level.

Reply:
@darranl  I have added new commits to both wildfly-elytron and wildfly-core PRs that make use of new interface for this reason. I have used separate commits so that if you will end up preferring a different solution or a different interface you can easily remove it. I did not manage to do a refactoring of SSLDefinitions class though

Reply:
@darranl Maybe we can remove the method from the new interface and leave the interface empty, and only return `false` here instead of `((ActiveSessionsSSLContext) sslContext).hasActiveSessions();`